src/main/java/io/cos/cas/osf/web/flow/login/OsfPrincipalFromNonInteractiveCredentialsAction.java

```diff
@@ -557,46 +558,55 @@ private OsfApiInstitutionAuthenticationResult notifyOsfApiOfInstnAuthnSuccess(
         }
         if (!isMemberOf.isEmpty()) {
             LOGGER.info(
-                    "[CAS XSLT] Secondary institution detected. SSO is '{}' and member is '{}'",
+                    "[CAS XSLT] Secondary institution detected: username={}, institution={}, member={}",
+                    username,
                     institutionId,
                     isMemberOf
             );
+        } else {
+            LOGGER.debug("[CAS XSLT] Secondary institution is not provided: username={}, institution={}", username, institutionId);
         }
-        final String payload = normalizedPayload.toString();
-        LOGGER.info(
-                "[CAS XSLT] All attributes checked: username={}, institution={}, member={}",
-                username,
-                institutionId,
-                isMemberOf
-        );
-        LOGGER.debug(
-                "[CAS XSLT] All attributes checked: username={}, institution={}, member={}, normalizedPayload={}",
-                username,
-                institutionId,
-                isMemberOf,
-                payload
-        );
-
+        // Parse the department attribute
         final String departmentRaw = user.optString("departmentRaw").trim();
         // Unlike all the above attributes that are released to us from the institutions, the `eduPerson` is a boolean
         // per-institution flag set by CAS in the "institutions-auth.xsl" file. It determines whether the department
         // attribute uses the the eduPerson schema (https://wiki.refeds.org/display/STAN/eduPerson).
         final boolean eduPerson = user.optBoolean("eduPerson");
-
         String department = "";
-        if (departmentRaw.isEmpty()) {
-            LOGGER.warn("[CAS XSLT] Missing department: fullname={}, username={}, institution={}", fullname, username, institutionId);
```

Comment:
`fullname` can be empty and it doesn't add value to the logging even when it is not.